novice-goals.Rmd

```diff
@@ -0,0 +1,201 @@
+
+-   What is the difference between validation and verification?
+    -   did we build the right thing vs. did we build it correctly
+-   How can I make my programs tell me that something has gone wrong?
+    -   assertions
+-   How can I write and manage tests for my software?
+    -   unit tests
+    -   test runners
+-   What tests should I write for my software?
+    -   edge cases
+    -   fails when it should
+    -   bugs become tests
+-   What *isn't* included?
+    -   legal liabilities
+
+## Version control
+
+-   What is a version control system?
+    -   a smarter kind of backup
+-   How do I track my work locally?
+    -   diff
+    -   add
+    -   commit
+    -   log
+-   How do I view or recover an old version of a file?
+    -   diff
+    -   checkout
+-   How do I save work remotely?
+    -   push and pull
+-   How do I manage conflicts?
+    -   merge
+-   What *isn't* included?
+    -   forking
+    -   branching
+    -   pull requests
+    -   git reflow --substantive --single-afferent-cycle --ia-ia-rebase-fhtagn ...
+
+## Publishing
+
+-   How do static websites work?
+    -   URLs
+    -   servers
+    -   request/response cycle
+    -   pages
+-   How do I create a simple HTML page?
+    -   head/body
+    -   basic elements
+    -   images
+    -   links
+    -   relative vs. absolute paths
+-   How can I create a simple website?
+    -   GitHub pages
+-   How can I give pages a standard appearance?
+    -   layouts
+-   How can I avoid writing all those tags?
+    -   Markdown
+-   How can I share values between pages?
+    -   flat per-site and per-page configuration
+    -   variable expansion
+-   What *isn't* included?
+    -   templating
+    -   filters
+    -   inclusions
+
+## Reproducibility
+
+-   How can I make programs easy to read?
+    -   coding style
+    -   documentation
+-   How can I make programs easy to re-use?
+    -   writing functions
+    -   Taschuk's Rules
+-   How can I combine explanations, code, and results?
+    -   notebooks
+-   How should I organize my projects?
+    -   Noble's Rules
+-   How should I keep track of my data?
+    -   simple manifests
+-   What *isn't* included?
+    -   build tools (Make and its kin)
+    -   continuous integration
+    -   documentation generators
+
+## Collaboration
+
+-   What kinds of licenses are there?
+    -   open vs. closed
+    -   copyright
+-   What license should I use for my publications?
+    -   CC-something
+-   What license should I use for my software?
+    -   MIT/BSD vs. GPL
+-   What license should I use for my data?
+    -   CC0
+-   How should I identify myself and my work?
+    -   DOIs
+    -   ORCIDs
+-   What's the difference between open and welcoming?
+    -   evidence for systematic exclusion
+    -   mechanics of exclusion
+-   How can I help create a level playing field?
+    -   what's wrong with deficit models
+    -   allyship
+    -   Code of Conduct (remove negatives)
+    -   kerb cuts (add positives)
```

Comment:
Yo, Greg, wuzzis?

Reply:
switched to simplified (American) spelling and added a link